### PR TITLE
Pointer function arguments

### LINKED ISSUE #26: pointer-function-arguments C tutorial
Populate C file with a tutorial and challenge.

### PR DESCRIPTION
**Pull request by Satantago**
### 2. Description
This tutorial explains function pointers as arguments in C. It first describes what a function pointer is and then shows how to pass a function pointer as a parameter to a function.
### 3. Fill in checklist by marking [x]

- [X] I've read the CONTRIBUTING.md
- [x] I was assigned to this issue
- [X] My code is formatted using required linting
- [X] I've run my code locally and checked it works

### CODE REVIEW

**Comment by funbeedev** [comment]
Thanks for the PR :) It will be reviewed in due course.

File: 08-pointers/pointer-function-arguments.c [hunk 1]
```diff
@@ -1,15 +1,91 @@
...
+(return_type) (*point_name) (parameter1, parameter2, ...);
+
+In the above syntax:
+  -> return_type: the variable type which is returned by the function
+  -> *pointer_name: the function pointer
+  -> parameter1, parameter2, ... : the list of parameters passed to the function
+
+e.g:
+  float (*add)();     this is a valid declaration for the function pointer
+  float *add();       this is an illegal declaration for the funciton pointer
```
Comment (sabderemane) [inline 08-pointers/pointer-function-arguments.c:21]:
```suggestion
  float *add();       this is an illegal declaration for the function pointer
```

File: 08-pointers/pointer-function-arguments.c [hunk 2]
```diff
@@ -1,15 +1,91 @@
...
+    of arguments in order. Let's call this function Function2
+ 3. We declare a pointer function of type TYPE1. Let's call it FunctionPointer.
+ 4. We assign Function2 to FunctionPointer:   FunctionPointer = Function2
+ 5. We then call function F1 with FunctionPointer as a parameter. 
+
+We shall see an illustration of this in the example below.
 ------------------------------------------------------------------------------------
 */
 
 // Code here explaining concept with comments to guide
```
Comment (sabderemane) [inline 08-pointers/pointer-function-arguments.c:39]:
```suggestion
```

File: 08-pointers/pointer-function-arguments.c [hunk 3]
```diff
@@ -1,15 +1,91 @@
...
+    return 0;
 }
+
 /*
 ------------------------------------------------------------------------------------
- Challenge: list challenges to be completed here. minimum of one challenge per tutorial
+ Challenge: 
+ 1. Write a function that prints "I love pointers!"
+ 2. Declare a corresponding function pointer and assign the previous function to it.
+ 3. Write a second function with the correspoing function pointer type as parameter.
```
Comment (sabderemane) [inline 08-pointers/pointer-function-arguments.c:88]:
```suggestion
 3. Write a second function with the corresponding function pointer type as parameter.
```

**Review by sabderemane — COMMENTED** [review]
Hi, I made some suggestions :)

**Comment by funbeedev** [comment]
Adding hacktoberfest-accepted label. Will review and merge later on.

**Pull request closed without merging**